From 9d863972a81ac9eaf121e99f87d4744bcf5fd5d3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Matt=C3=A9o=20Rossillol=E2=80=91=E2=80=91Laruelle?=
 <beatussum@protonmail.com>
Date: Wed, 18 Oct 2023 19:31:02 +0200
Subject: [PATCH] fix build failure with LLVM/Clang 17
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The `str` macro defined in `src/version.h` seems to override some system
headers.

Signed-off-by: Mattéo Rossillol‑‑Laruelle <beatussum@protonmail.com>
---
 src/version.h | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/version.h b/src/version.h
index 656e1716..5c8d2841 100644
--- a/src/version.h
+++ b/src/version.h
@@ -4,8 +4,7 @@
 #define MINOR_VERSION 12
 #define PATCH_VERSION 2
 
-#define str(s) #s
-#define xstr(s) str(s)
+#define xstr(s) #s
 #define APP_VERSION xstr(MAJOR_VERSION) "." xstr(MINOR_VERSION) "." xstr(PATCH_VERSION)
 
 // If it is defined by the compiler, then it is a nightly build, and in the YYYYMMDD format.
-- 
2.41.0

